app/services/AdjacentCellsCalculator.coffee

```diff
@@ -1,25 +1,30 @@
-Tuple = require "../models/Tuple"
+Tuple    = require "../models/Tuple"
+TupleSet = require "../models/TupleSet"
 
 class AdjacentCellsCalculator
 
   # Param grid is a GameGrid
-  constructor: (@grid, @cells, @x, @y) ->
+  constructor: (@grid, @cells=(new TupleSet), @x, @y) ->
 
   # Gets the adjacent cells
-  calculate: () => 
-    for i in [@y - 1, @y, @y + 1]
-      for j in [@x - 1, @x, @x + 1]
-        continue if i is @y and j is @x
-        @cells.push @validLocation @grid, i, j
-    @cells
+  calculate: (takenCells) =>
+    for i in [@x - 1, @x, @x + 1]
+      for j in [@y - 1, @y, @y + 1]
+        continue if i is @x and j is @y
+        @cells.push @validLocation @grid, i, j, takenCells
+    @cells.list
 
   # returns a valid location if it exists, otherwise null
-  validLocation: (grid,x,y) =>
+  validLocation: (grid, x, y, takenCells) =>
     while grid.validIndices x, y
-      return new Tuple x, y if @empty grid, x, y
+      for cell in takenCells
+        if x is cell.x and y < cell.y
+          return null
+      return new Tuple x, y if (@empty grid, x, y)
```

Comment:
20-22 could be a separate method to improve readability. Barring that, it might be good to stick a comment about it. saying `# return null if the location is below a taken cell`

Also, I think you could reduce function calls if you change line 21 to test
`y < = cell.y `